pypesto/objective/base.py

```diff
@@ -527,7 +527,8 @@ def check_grad(
             data = {**prefix_data, **data, **postfix_data}
 
         # create dataframe
-        result = pd.DataFrame(data=data)
+        result = pd.DataFrame(data=data,
+                              index=[self.x_names[ix] for ix in x_indices])
```

Comment:
`self.x_names` defaults to `None`. Not sure if this will ever be the case here though.

Reply:
I think whenever parameter names are used, a good default are p0, ... / p_0, ... / x0, ..., or sth like that.

Reply:
cannot add default names to objective since dimension of x is not known at initialization